packages/express/src/__tests__/acceptance/invoke-middleware.acceptance.ts

```diff
@@ -54,6 +61,34 @@ describe('Express middleware registry', () => {
       .expect('x-spy-mock', 'POST /hello');
   });
 
+  it('invokes middleware using the provider', async () => {
+    const logMiddleware = toMiddleware(spyMiddlewareFactory({action: 'log'}));
+    const mockMiddleware = toMiddleware(spyMiddlewareFactory({action: 'mock'}));
+
+    server.middleware(logMiddleware, {chain: 'log'});
+    server.middleware(mockMiddleware, {chain: 'mock'});
+
+    const binding = createBindingFromClass(InvokeMiddlewareProvider);
+    server.add(binding);
+    const invoke = await server.get<InvokeMiddleware>(binding.key);
+
+    server.middleware(async (ctx, next) => {
+      return invoke(ctx, {
+        chain: 'log',
+        next: () => {
+          return invoke(ctx, {chain: 'mock', next});
```

Comment:
Had a quick chat with @raymondfeng , the `next` function will be honored as part of the `options`. 👍